lib/appium.js

```diff
@@ -172,9 +172,18 @@ class AppiumDriver extends BaseDriver {
     inspectObject(caps);
   }
 
-  async createSession (desiredCaps, reqCaps, capabilities) {
-    if (capabilities) {
-      desiredCaps = processCapabilities(capabilities, null, false);
+  /**
+   * Create a new session
+   * @param {Object} desiredCaps JSONWP formatted desired capabilities
+   * @param {Object} reqCaps Required capabilities
+   * @param {Object} capabilities W3C capabilities
+   */
+  async createSession (desiredCaps={}, reqCaps, capabilities) {
+    if (_.isObject(capabilities)) {
+      // Merging W3C caps into desiredCaps is a stop-gap until all the clients and drivers become fully W3C compliant
+      log.info(`Merged W3C capabilities ${_.truncate(JSON.stringify(capabilities), 20)} into desiredCapabiilities object ${JSON.stringify(desiredCaps)}`);
```

Comment:
is 20 chars enough?

Reply:
This will actually truncate to 30, the default, because the usage is `_.truncate(JSON.stringify(capabilities), {length: 20})`

Reply:
Type: `desiredCapabiilities`